zeebe/journal/src/main/java/io/camunda/zeebe/journal/file/SegmentsManager.java

```diff
@@ -279,18 +279,23 @@ private void resetCurrentSegment() {
   void open() {
     final var openDurationTimer = journalMetrics.startJournalOpenDurationTimer();
     // Load existing log segments from disk.
-    loadSegments()
-        .forEach(
-            segment -> {
-              final Segment replacedSegment = segments.put(segment.descriptor().index(), segment);
-              if (replacedSegment != null) {
-                // The previous segment can be safely deleted to free resources.
-                replacedSegment.close();
-                replacedSegment.delete();
-              } else {
-                journalMetrics.incSegmentCount();
-              }
-            });
+    Segment previousSegment = null;
+    for (final Segment segment : loadSegments()) {
+      if (previousSegment != null && previousSegment.lastIndex() >= segment.index()) {
+        // Segments are overlapping, remove the journal index entries in the segment's index range.
+        // This is to avoid having an index lookup point to a position in the previous segment.
+        journalIndex.deleteInRange(segment.index(), segment.lastIndex() - 1);
```

Comment:
🔧 This should only clear from `segment.index()` to `previousSegment.lastIndex()` because that's the range that is invalid. Entries between `previousSegment.lastIndex()` and `segment.lastIndex()` remain valid.